build.sbt

```diff
@@ -296,24 +300,40 @@ lazy val sparkNotebook = project.in(file(".")).enablePlugins(PlayScala).enablePl
   .dependsOn(sparkNotebookCore, gitNotebookProvider, subprocess, observable, sbtProjectGenerator, common, spark, kernel)
   .settings(sharedSettings: _*)
   .settings(
-    bashScriptExtraDefines <+= (organization, version, scalaBinaryVersion, scalaVersion, sparkVersion, hadoopVersion, withHive) map { (org, v, sbv, sv, pv, hv, wh) =>
-      s"""export ADD_JARS="$${ADD_JARS},$${lib_dir}/$$(ls $${lib_dir} | grep ${org}.common | head)""""
-    },
+    bashScriptExtraDefines +=
+      s"""export ADD_JARS="$${ADD_JARS},$${lib_dir}/$$(ls $${lib_dir} | grep ${organization.value}.common | head)"""",
+
     // configure the "universal" distribution package (i.e. spark-notebook.zip)
     mappings in Universal ++= directory("notebooks"),
-    version in Universal <<= (version in ThisBuild, scalaVersion, sparkVersion, hadoopVersion, withHive) { (v, sc, sv, hv, h) =>
-                                s"$v-scala-$sc-spark-$sv-hadoop-$hv" + (if (h) "-with-hive" else "")
-                              },
-    version in Docker <<= (version in ThisBuild, scalaVersion, sparkVersion, hadoopVersion, withHive) { (v, sc, sv, hv, h) =>
-                                s"$v-scala-$sc-spark-$sv-hadoop-$hv" + (if (h) "-with-hive" else "")
-                              },
-    version in Debian <<= (version in ThisBuild, scalaVersion, sparkVersion, hadoopVersion, withHive) { (v, sc, sv, hv, h) =>
-                                s"$v-scala-$sc-spark-$sv-hadoop-$hv" + (if (h) "-with-hive" else "")
-                              },
+
+    version in Universal := Seq(
+      s"${version.in(ThisBuild).value}",
+      s"-scala-${scalaVersion.value}",
+      s"-spark-${sparkVersion.value}",
+      s"-hadoop-${hadoopVersion.value}",
+      if(withHive.value) "-with-hive" else ""
+    ).mkString,
+
+    version in Docker := Seq(
+      s"${version.in(ThisBuild).value}",
+      s"-scala-${scalaVersion.value}",
+      s"-spark-${sparkVersion.value}",
+      s"-hadoop-${hadoopVersion.value}",
+      if(withHive.value) "-with-hive" else ""
+    ).mkString,
+
+    version in Debian := Seq(
+      s"${version.in(ThisBuild).value}",
+      s"-scala-${scalaVersion.value}",
+      s"-spark-${sparkVersion.value}",
+      s"-hadoop-${hadoopVersion.value}",
+      if(withHive.value) "-with-hive" else ""
+    ).mkString,
```

Comment:
should exract this as a helper fn, as it's repeated 3 times...

Reply:
I was thinking about that as well. Will do...